### PR TITLE
Cherry pick #482 from cloud provider azure: aggregate errors when putting vmss

### LINKED ISSUE #481: Not all node pools are cleaned up on ILB deletion
**What happened**:

Because the error is returned on each VMSS update, if there're errors happen on first VMSS, then all the following VMSS won't be updated:

https://github.com/kubernetes/legacy-cloud-providers/blob/785692fc4071ffcc2647a3fb8c2c22294db36de6/azure/azure_vmss.go#L1466-L1470
image

An example of error is:

```sh
Operation 'Update Virtual Machine Scale Set' is not allowed on Virtual Machine Scale Set 'aks-xxx-vmss' since it is marked for deletion.
```

**What you expected to happen**:

1. Cloud provider should save the error for each VMSS into a list and return an aggregated error at the end.
2. Cloud provider should avoid PUT on deleting VMSS.

**How to reproduce it**:

After fixing the issue on 1.21, we should also cherry pick the fix to older versions.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

/milestone v1.21

### PR DESCRIPTION
**Pull request by nilo19**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:
https://github.com/kubernetes-sigs/cloud-provider-azure/pull/482/commits: Aggregate errors when putting vmss, so the failure of the previous update would not interrupt the next ones.

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes-sigs/cloud-provider-azure/issues/481

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
aggregate errors when putting vmss
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```
/priority critical-urgent
/triage accepted
/sig cloud-provider
/area provider/azure

### CODE REVIEW

**Review by feiskyer — COMMENTED** [review]
/lgtm
/approve

**Comment by feiskyer** [comment]
@andrewsykim could you help to approve this cherry pick?

/assign andrewsykim

**Comment by feiskyer** [comment]
/test pull-kubernetes-unit

ping @andrewsykim @cheftako for approval

**Comment by nilo19** [comment]
ping @andrewsykim @cheftako for approval

**Comment by cheftako** [comment]
A little confusing labelling this as a cherry pick. I realize your forklifting a fix from cloud provider azure, but generally when we refer to cherry picks we are referring to backporting fixes within K/K to an older branch, rather than importing fixes from another repo.

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss.go [hunk 1]
```diff
@@ -1431,10 +1431,15 @@ func (ss *scaleSet) ensureBackendPoolDeletedFromVMSS(service *v1.Service, backen
 		vmssNamesMap[vmSetName] = true
 	}
 
+	vmssUpdaters := make([]func() error, 0, len(vmssNamesMap))
+	errors := make([]error, 0, len(vmssNamesMap))
 	for vmssName := range vmssNamesMap {
+		vmssName := vmssName
```
Comment (cheftako) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss.go:1437]: Might be good to add a comment explaining why you are shadowing this variable.

**Review by feiskyer — COMMENTED** [review]
/lgtm
/retest

**Pull request merged**